tests/contrib/celery/test_task.py

```diff
@@ -459,41 +414,19 @@ def test_task_delay_eager(self):
         # DEV: Assert as endswith, since PY3 gives us `u'is_eager` and PY2 gives us `'is_eager'`
         self.assertTrue(meta['celery.delivery_info'].endswith('\'is_eager\': True}'))
 
-    def test_apply_async_previous_style_tasks(self):
-        # ensures apply_async is properly patched if Celery 1.0 style tasks
-        # are used even in newer versions. This should extend support to
-        # previous versions of Celery.
-        # Regression test: https://github.com/DataDog/dd-trace-py/pull/449
-        app = celery.Celery('test_task_delay_eager', broker=self.broker_url)
-        app.conf['CELERY_ALWAYS_EAGER'] = True
-
-        class CelerySuperClass(celery.task.Task):
-            abstract = True
-
-            @classmethod
-            def apply_async(cls, args=None, kwargs=None, **kwargs_):
-                return super(CelerySuperClass, cls).apply_async(args=args, kwargs=kwargs, **kwargs_)
-
-            def run(self, *args, **kwargs):
-                if 'stop' in kwargs:
-                    # avoid call loop
-                    return
-                CelerySubClass.apply_async(args=[], kwargs={"stop": True})
-
-        @patch_task_with_pin(pin=self.pin)
-        class CelerySubClass(CelerySuperClass):
-            pass
-
-        t = CelerySubClass()
-        t.run()
-        spans = self.tracer.writer.pop()
-        self.assertEqual(len(spans), 4)
-
     def test_celery_shared_task(self):
+        # Ensure Django Shared Task are supported
         @celery.shared_task
         def add(x ,y):
             return x + y
 
+        # TODO[manu]: this should not happen. We're not propagating the `Pin`
+        # from the main app and so it's difficult to change globally (or per `Task`)
+        # our tracing configurations. After solving the Pin propagation, remove
+        # this `Pin.override`.
+        # Probably related to: https://github.com/DataDog/dd-trace-py/issues/510
+        Pin.override(add, tracer=self.tracer)
```

Comment:
This is probably related to a bug where we cannot change the service name because we're not propagating the app `Pin` to all tasks.